qermit/spam/spam_mitres.py

```diff
@@ -101,6 +101,10 @@ def gen_UnCorrelated_SPAM_MitRes(
     """
     if backend.backend_info is None:
         raise ValueError("Backend has no backend_info attribute.")
+    if backend.backend_info.architecture is None:
+        raise ValueError(
+            "Backend Architecture has no specified Nodes, please use a Backend with a specified Architecture."
```

Comment:
I'd change the error message to:
`BackendInfo stored by Backend has no defined Architecture, please use a Backend with a specified Architecture.`